### PR TITLE
[CI Visibility] - Add a final try to load the pdb from a module.

### PR DESCRIPTION
**Pull request by tonyredondo**
## Summary of changes

This PR adds a fallback when the `MethodSymbolResolver` cannot open the PDB file automatically, by manually trying to find the pdb as a last resort.

## Reason for change

When working in the test logger, this scenario appears to happen frequently, this workaround fixes the issue.

### CODE REVIEW

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #3350 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13): 
* 1 benchmarks are **slower**, with geometric mean 1.130
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`WriteAndFlushEnrichedTraces`|net472|782μs|903ns|3.38μs|0.388| 0|0|3.22&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|511μs|214ns|770ns|0| 0|0|2.62&nbsp;KB|
|#3350|`WriteAndFlushEnrichedTraces`|net472|784μs|1.48μs|5.73μs|0.393| 0|0|3.22&nbsp;KB|
|#3350|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|525μs|131ns|454ns|0| 0|0|2.62&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AppSecBodyBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #3350

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.AppSecBodyBenchmark.AllCycleMoreComplexBody&#8209;net472 | 1.130 | 194.25 | 219.44 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`AllCycleSimpleBody`|net472|196ns|0.0987ns|0.382ns|0.0727| 0|0|457&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`AllCycleSimpleBody`|netcoreapp3.1|256ns|0.257ns|0.994ns|0.00635| 0|0|456&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`AllCycleMoreComplexBody`|net472|194ns|0.0439ns|0.158ns|0.0688| 0|0|433&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`AllCycleMoreComplexBody`|netcoreapp3.1|253ns|0.22ns|0.852ns|0.00594| 0|0|432&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`BodyExtractorSimpleBody`|net472|295ns|0.157ns|0.609ns|0.0573| 0|0|361&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`BodyExtractorSimpleBody`|netcoreapp3.1|239ns|0.554ns|2ns|0.00378| 0|0|272&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`BodyExtractorMoreComplexBody`|net472|16.2μs|7.76ns|29ns|1.2| 0.0163|0|7.62&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`BodyExtractorMoreComplexBody`|netcoreapp3.1|12.9μs|5.6ns|19.4ns|0.0903| 0|0|6.75&nbsp;KB|
|#3350|`AllCycleSimpleBody`|net472|198ns|0.102ns|0.395ns|0.0727| 0|0|457&nbsp;B|
|#3350|`AllCycleSimpleBody`|netcoreapp3.1|256ns|0.117ns|0.451ns|0.00618| 0|0|456&nbsp;B|
|#3350|`AllCycleMoreComplexBody`|net472|220ns|0.439ns|1.7ns|0.0689| 0|0|433&nbsp;B|
|#3350|`AllCycleMoreComplexBody`|netcoreapp3.1|257ns|0.332ns|1.29ns|0.00596| 0|0|432&nbsp;B|
|#3350|`BodyExtractorSimpleBody`|net472|277ns|0.257ns|0.994ns|0.0573| 0|0|361&nbsp;B|
|#3350|`BodyExtractorSimpleBody`|netcoreapp3.1|238ns|0.128ns|0.497ns|0.00372| 0|0|272&nbsp;B|
|#3350|`BodyExtractorMoreComplexBody`|net472|15.7μs|15.2ns|58.8ns|1.21| 0.0158|0|7.62&nbsp;KB|
|#3350|`BodyExtractorMoreComplexBody`|netcoreapp3.1|13.1μs|4.35ns|15.7ns|0.0918| 0|0|6.75&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Unknown :shrug: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`SendRequest`|netcoreapp3.1|182μs|281ns|1.09μs|0.271| 0|0|20.68&nbsp;KB|
|#3350|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|#3350|`SendRequest`|netcoreapp3.1|180μs|267ns|1.03μs|0.18| 0|0|20.68&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`ExecuteNonQuery`|net472|1.85μs|0.549ns|2.05ns|0.155| 0.000925|0|979&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`ExecuteNonQuery`|netcoreapp3.1|1.36μs|0.65ns|2.52ns|0.0129| 0|0|968&nbsp;B|
|#3350|`ExecuteNonQuery`|net472|1.84μs|0.928ns|3.6ns|0.155| 0.000921|0|979&nbsp;B|
|#3350|`ExecuteNonQuery`|netcoreapp3.1|1.42μs|0.685ns|2.65ns|0.0129| 0|0|968&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`CallElasticsearch`|net472|2.54μs|0.926ns|3.46ns|0.189| 0|0|1.19&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`CallElasticsearch`|netcoreapp3.1|1.52μs|0.838ns|3.25ns|0.0152| 0|0|1.13&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`CallElasticsearchAsync`|net472|2.64μs|0.838ns|3.14ns|0.21| 0|0|1.32&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`CallElasticsearchAsync`|netcoreapp3.1|1.63μs|0.749ns|2.9ns|0.017| 0|0|1.25&nbsp;KB|
|#3350|`CallElasticsearch`|net472|2.66μs|9.43ns|36.5ns|0.188| 0|0|1.19&nbsp;KB|
|#3350|`CallElasticsearch`|netcoreapp3.1|1.61μs|0.597ns|2.31ns|0.0154| 0|0|1.13&nbsp;KB|
|#3350|`CallElasticsearchAsync`|net472|2.78μs|1.82ns|7.05ns|0.21| 0|0|1.32&nbsp;KB|
|#3350|`CallElasticsearchAsync`|netcoreapp3.1|1.57μs|0.501ns|1.87ns|0.0164| 0|0|1.25&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`ExecuteAsync`|net472|2.73μs|1.49ns|5.77ns|0.23| 0|0|1.45&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`ExecuteAsync`|netcoreapp3.1|1.8μs|0.466ns|1.74ns|0.019| 0|0|1.38&nbsp;KB|
|#3350|`ExecuteAsync`|net472|2.77μs|0.673ns|2.61ns|0.231| 0|0|1.45&nbsp;KB|
|#3350|`ExecuteAsync`|netcoreapp3.1|1.76μs|0.656ns|2.45ns|0.0194| 0|0|1.38&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`SendAsync`|net472|5.91μs|2.14ns|8.29ns|0.445| 0|0|2.8&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`SendAsync`|netcoreapp3.1|3.78μs|2.17ns|8.42ns|0.0359| 0|0|2.63&nbsp;KB|
|#3350|`SendAsync`|net472|5.87μs|1.59ns|5.93ns|0.443| 0|0|2.8&nbsp;KB|
|#3350|`SendAsync`|netcoreapp3.1|3.69μs|1.28ns|4.78ns|0.0351| 0|0|2.63&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`EnrichedLog`|net472|3.13μs|3.3ns|12.8ns|0.298| 0|0|1.88&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`EnrichedLog`|netcoreapp3.1|2.59μs|1.41ns|5.47ns|0.0259| 0|0|1.91&nbsp;KB|
|#3350|`EnrichedLog`|net472|3.25μs|3.21ns|12ns|0.297| 0|0|1.88&nbsp;KB|
|#3350|`EnrichedLog`|netcoreapp3.1|2.51μs|1.51ns|5.44ns|0.0257| 0|0|1.91&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`EnrichedLog`|net472|151μs|161ns|624ns|0.679| 0.226|0|4.72&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`EnrichedLog`|netcoreapp3.1|120μs|160ns|620ns|0.0598| 0|0|4.55&nbsp;KB|
|#3350|`EnrichedLog`|net472|155μs|254ns|983ns|0.689| 0.23|0|4.72&nbsp;KB|
|#3350|`EnrichedLog`|netcoreapp3.1|121μs|187ns|724ns|0.0604| 0|0|4.55&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`EnrichedLog`|net472|5.79μs|3.05ns|11ns|0.578| 0.00289|0|3.65&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`EnrichedLog`|netcoreapp3.1|4.66μs|2.04ns|7.88ns|0.053| 0|0|3.98&nbsp;KB|
|#3350|`EnrichedLog`|net472|5.64μs|2.34ns|9.04ns|0.578| 0.00282|0|3.65&nbsp;KB|
|#3350|`EnrichedLog`|netcoreapp3.1|4.38μs|1.41ns|5.45ns|0.055| 0|0|3.98&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`SendReceive`|net472|2.34μs|3.17ns|12.3ns|0.222| 0|0|1.4&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`SendReceive`|netcoreapp3.1|1.88μs|0.453ns|1.63ns|0.0188| 0|0|1.35&nbsp;KB|
|#3350|`SendReceive`|net472|2.36μs|2.17ns|8.41ns|0.223| 0|0|1.4&nbsp;KB|
|#3350|`SendReceive`|netcoreapp3.1|1.84μs|0.922ns|3.57ns|0.0185| 0|0|1.35&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`EnrichedLog`|net472|5.07μs|2.69ns|10.4ns|0.364| 0|0|2.3&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`EnrichedLog`|netcoreapp3.1|4.37μs|1.2ns|4.66ns|0.024| 0|0|1.86&nbsp;KB|
|#3350|`EnrichedLog`|net472|5.03μs|4.01ns|15.5ns|0.362| 0|0|2.3&nbsp;KB|
|#3350|`EnrichedLog`|netcoreapp3.1|4.44μs|2.02ns|7.27ns|0.0245| 0|0|1.86&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`StartFinishSpan`|net472|1.16μs|0.296ns|1.11ns|0.133| 0|0|842&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`StartFinishSpan`|netcoreapp3.1|960ns|0.283ns|1.06ns|0.0106| 0|0|792&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`StartFinishScope`|net472|1.37μs|0.393ns|1.47ns|0.146| 0|0|923&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`StartFinishScope`|netcoreapp3.1|1.09μs|1.38ns|5.14ns|0.0124| 0|0|912&nbsp;B|
|#3350|`StartFinishSpan`|net472|1.18μs|0.574ns|2.07ns|0.133| 0|0|842&nbsp;B|
|#3350|`StartFinishSpan`|netcoreapp3.1|988ns|0.784ns|2.83ns|0.0108| 0|0|792&nbsp;B|
|#3350|`StartFinishScope`|net472|1.47μs|0.645ns|2.41ns|0.146| 0|0|923&nbsp;B|
|#3350|`StartFinishScope`|netcoreapp3.1|1.08μs|0.795ns|3.08ns|0.0122| 0|0|912&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`RunOnMethodBegin`|net472|1.55μs|0.474ns|1.84ns|0.147| 0|0|923&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/b08adc4ffd11d2045fab9c6de30ff7adabe0fd13)|`RunOnMethodBegin`|netcoreapp3.1|1.16μs|0.222ns|0.83ns|0.0122| 0|0|912&nbsp;B|
|#3350|`RunOnMethodBegin`|net472|1.53μs|0.671ns|2.6ns|0.147| 0|0|923&nbsp;B|
|#3350|`RunOnMethodBegin`|netcoreapp3.1|1.21μs|0.319ns|1.24ns|0.0121| 0|0|912&nbsp;B|
</details>

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #3350 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/6a8b825c22aa41752267793f3c3f4a6816ce014a) will **not change** line coverage
:heavy_check_mark: Merging #3350 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/6a8b825c22aa41752267793f3c3f4a6816ce014a) will **not change** branch coverage
:no_entry: Merging #3350 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/6a8b825c22aa41752267793f3c3f4a6816ce014a) will will **increase** complexity by `6`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/6a8b825c22aa41752267793f3c3f4a6816ce014a) | #3350       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `19534` / `26924` | `19521` / `26929` |          |
| Lines %   | `73%`      | `72%`      |  `0%` :heavy_check_mark:  |
| Branches  | `11607` / `16747` | `11601` / `16753` |          |
| Branches %| `69%`      | `69%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `18113`      | `18119`        |  `6`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=115220&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/115220/artifacts?artifactName=Code%20Coverage%20Report_115220&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3350](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=115235&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/115235/artifacts?artifactName=Code%20Coverage%20Report_115235&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/6a8b825c22aa41752267793f3c3f4a6816ce014a) | #3350       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `73%`      | `72%`       |  `0%` :heavy_check_mark:  |
| Branches %| `69%`      | `69%`       |  `0%` :heavy_check_mark:  |
| Complexity| `18113`      | `18119`       |  `6` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Ci.GitInfo](https://github.com/DataDog/dd-trace-dotnet/tree/4f4356e78bc69840e2b088ec2717407545bd6506/tracer/src/Datadog.Trace/Ci/GitInfo.cs) | `5%` :heavy_check_mark: | `8%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Propagators.SpanContextPropagator](https://github.com/DataDog/dd-trace-dotnet/tree/4f4356e78bc69840e2b088ec2717407545bd6506/tracer/src/Datadog.Trace/Propagators/SpanContextPropagator.cs) | `8%` :heavy_check_mark: | `6%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=115220&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/115220/artifacts?artifactName=Code%20Coverage%20Report_115220&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3350](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=115235&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/115235/artifacts?artifactName=Code%20Coverage%20Report_115235&api-version=6.0&%24format=zip)

**Review by andrewlock — APPROVED** [review]

File: tracer/src/Datadog.Trace/PDBs/MethodSymbolResolver.cs [hunk 1]
```diff
@@ -178,6 +180,16 @@ public bool TryGetMethodSymbol(MethodInfo methodInfo, out MethodSymbol methodSym
...
                             moduleDef = mDef;
                         }
+                        else if (options.PdbFileOrData is null && !string.IsNullOrEmpty(module.Assembly.Location))
+                        {
+                            // If the PDB cannot be loaded automatically we do a final try to manually find it.
+                            var pdbFile = Path.ChangeExtension(module.Assembly.Location, "pdb");
+                            if (File.Exists(pdbFile))
+                            {
+                                options.PdbFileOrData = pdbFile;
+                                goto loadModule;
```
Comment (lucaspimentel) [inline tracer/src/Datadog.Trace/PDBs/MethodSymbolResolver.cs:190]: `goto` 👀

**Review by lucaspimentel — APPROVED** [review]

File: tracer/src/Datadog.Trace/PDBs/MethodSymbolResolver.cs (hunk 1 above)
Comment (tonyredondo) [reply tracer/src/Datadog.Trace/PDBs/MethodSymbolResolver.cs:190]: It’s time to embrace it 🤣

**Review by zacharycmontoya — APPROVED** [review]

File: tracer/src/Datadog.Trace/PDBs/MethodSymbolResolver.cs [hunk 2]
```diff
@@ -178,6 +180,16 @@ public bool TryGetMethodSymbol(MethodInfo methodInfo, out MethodSymbol methodSym
                         {
                             moduleDef = mDef;
                         }
+                        else if (options.PdbFileOrData is null && !string.IsNullOrEmpty(module.Assembly.Location))
+                        {
+                            // If the PDB cannot be loaded automatically we do a final try to manually find it.
+                            var pdbFile = Path.ChangeExtension(module.Assembly.Location, "pdb");
```
Comment (lucaspimentel) [inline tracer/src/Datadog.Trace/PDBs/MethodSymbolResolver.cs:186]: From [the docs](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.location?view=net-6.0#remarks):

> .NET Framework only: If the loaded file was [shadow-copied](https://learn.microsoft.com/en-us/dotnet/framework/app-domains/shadow-copy-assemblies), the location is that of the file after being shadow-copied. To get the location before the file has been shadow-copied, use the [CodeBase](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.codebase?view=net-6.0) property.

Comment (lucaspimentel) [reply tracer/src/Datadog.Trace/PDBs/MethodSymbolResolver.cs:186]: But also from [the docs for `Assembly.CodeBase`](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.codebase?view=net-6.0):

> `Assembly.CodeBase` and `Assembly.EscapedCodeBase` are only included for .NET Framework compatibility. Use `Assembly.Location`.

😅

Comment (tonyredondo) [reply tracer/src/Datadog.Trace/PDBs/MethodSymbolResolver.cs:186]: image

**Review by OmerRaviv — COMMENTED** [review]
LGTM

**Pull request merged**